File: crush-algorithm-go/src/cluster.go
package main

import "fmt"

// 打印集群节点信息
func PrintCluster(cluster *Cluster) {
	fmt.Printf("Cluster contains %d nodes:\n", cluster.NodeCount)
	for _, node := range cluster.Nodes {
		fmt.Printf("Node %d: %s\n", node.ID, node.Name)
	}
}

File: crush-algorithm/src/cluster.c
#include "cluster.h"

// 打印集群节点信息
void print_cluster(const Cluster* cluster) {
    printf("Cluster contains %d nodes:\n", cluster->node_count);
    for (int i = 0; i < cluster->node_count; i++) {
        printf("Node %d: %s\n", cluster->nodes[i].id, cluster->nodes[i].name);
    }
}
